File: js/binary_search/binary_search.js
const arr = [10, 20, 30, 50, 60, 80, 110, 130, 140, 170];
const X = 110;

const binarySearch = (arr, X) => {
  var start = 0,
    end = arr.length - 1,
    mid;

  while (start <= end) {
    mid = Math.floor((start + end) / 2);
    if (arr[mid] == X) {
      return true;
    } else if (arr[mid] < X) {
      start = mid + 1;
    } else {
      end = mid - 1;
    }
  }
  return false;
};

// const resultStatus = binarySearch(arr, X);
// console.log(`The value ${X} in the arr is: ${resultStatus}`);

module.exports = binarySearch;

File: init17/recusion/binary_search.py
def binarySearch(li, item, end, start=0):

    if end < start:
        return False
    mid = (end+start)//2

    if li[mid]==item:
        return mid+1
    if li[mid] < item:
        return binarySearch(li, item, end, mid+1)
    if li[mid] > item:
        return binarySearch(li, item, mid-1, start)

li = [1,2,3,4,5,6,7,8,9]
item = 7
pr = binarySearch(li, item, 8)
if pr:
    print('%d found on %d position' %(item, pr))
else:
    print('%d is not found on the list' %(item))
